src/main/java/eu/bbmri_eric/negotiator/settings/UIParameterValidator.java

```diff
@@ -0,0 +1,30 @@
+package eu.bbmri_eric.negotiator.settings;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class UIParameterValidator implements ConstraintValidator<ValidUIParameter, UIParameter> {
+
+  @Override
+  public void initialize(ValidUIParameter constraintAnnotation) {}
+
+  @Override
+  public boolean isValid(UIParameter parameter, ConstraintValidatorContext context) {
+    switch (parameter.getType()) {
+      case BOOL:
+        return parameter.getValue().equalsIgnoreCase("true")
+            || parameter.getValue().equalsIgnoreCase("false");
+      case INT:
+        try {
+          Integer.parseInt(parameter.getValue());
+          return true;
+        } catch (NumberFormatException e) {
+          return false;
+        }
+      case STRING:
+        return true;
+      default:
+        return false;
```

Comment:
isn't this wrong separation of concerns? If you add a new type then you have to change code in 3 places

Reply:
The only concern of this class is to check whether the parameter's value is compatible with the declared type. This is the way I found to validate an Entity instance at the class level (which is the case since the validation of the value depends on the type). 
I removed the problem of changing the service when there is a new type, but this class is by definition coupled with the entity (it's its validator), so I would keep it as it is.